src/charm.py

```diff
@@ -224,10 +224,16 @@ def _check_precondition(self) -> None:
             )
         nginx_route_relation = self.model.get_relation("nginx-route")
         ingress_relation = self.model.get_relation("ingress")
+        # The charm only supports one relation at a time, so we check if both are present
+        # and raise an error if they are.
         if nginx_route_relation is not None and ingress_relation is not None:
             raise InvalidIngressError(
                 "nginx-ingress-integrator cannot establish more than one relation at a time"
```

Comment:
I guess out of scope for this PR but I think the error message could be slightly better phrased as an actionable item: such as "Both Nginx relation and Ingress relation found. Please remove either one"

Reply:
i can quickly make this change :)